### ISSUE TITLE
Seems the hooks have some problems

### ISSUE THREAD

**Issue by wolfshow** [google/seq2seq#167]
I use the train_seq2seq.yml and evaluate the model every 1000 steps. But once it starts to evaluate, the process will crash with the following information. I am not sure whether there is something wrong with the hooks.

`InvalidArgumentError (see above for traceback): logits and labels must have the same first dimension, got logits shape [1600,37007] and labels shape [3136]
	 [[Node: model/att_seq2seq/cross_entropy_sequence_loss/SparseSoftmaxCrossEntropyWithLogits/SparseSoftmaxCrossEntropyWithLogits = SparseSoftmaxCrossEntropyWithLogits[T=DT_FLOAT, Tlabels=DT_INT64, _device="/job:localhost/replica:0/task:0/gpu:0"](model/att_seq2seq/cross_entropy_sequence_loss/SparseSoftmaxCrossEntropyWithLogits/Reshape, model/att_seq2seq/cross_entropy_sequence_loss/SparseSoftmaxCrossEntropyWithLogits/Reshape_1)]]
	 [[Node: mean/broadcast_weights/assert_broadcastable/is_valid_shape/has_valid_nonscalar_shape/has_invalid_dims/concat/_567 = _Recv[client_terminated=false, recv_device="/job:localhost/replica:0/task:0/cpu:0", send_device="/job:localhost/replica:0/task:0/gpu:0", send_device_incarnation=1, tensor_name="edge_575_mean/broadcast_weights/assert_broadcastable/is_valid_shape/has_valid_nonscalar_shape/has_invalid_dims/concat", tensor_type=DT_INT32, _device="/job:localhost/replica:0/task:0/cpu:0"]()]]`

**Comment by wolfshow** [comment]
after remove the time sample ouput. got the following error.

`W tensorflow/core/framework/op_kernel.cc:993] Invalid argument: Tried to read from index 49 but array size is: 49`

**Comment by dennybritz** [comment]
Related to #103 - Try to pull the latest version. Please re-open if you still have this problem.
